### DISCUSSION TITLE
Improve Entry Date field to make it user friendly to pick a different time

### DISCUSSION THREAD

**Discussion by studiopatagon** [ExpressionEngine/ExpressionEngine#3021] [Feature Requests]
The current field for entry date and entry expiration date presents a problem to some editors where they have to manually edit the date/time string to set a different time for the entry. 

The solution presented makes it more user friendly and error proof while still allowing those who want to edit the string to do so.

<img width="333" alt="timeruff" src="https://user-images.githubusercontent.com/12139134/219406212-b5ef0c93-9457-4498-891a-a4357610843e.png">

**Comment by blairliikala** [comment]
Are there any existing date/time projects that could be used here?  Would it be helpful to change the native `input` element to `<input type="datetime-local">`?

**Reply by studiopatagon** [reply]
I think that's a good idea. I'm not aware of any off the shelf implementation for this. I think it'd be fairly easy to have the time at the bottom update the string with some simple JS.

**Comment by bsrweb** [comment]
+1 to this.

**Comment by jaredchesebro** [comment]
This looks great. I would also consider adding +- or up and down arrows to hours and minutes inputs.

**Comment by pyrobob83** [comment]
Another +1 from me in terms of making it easier for users to set a specific time - I think this looks great and I like the idea of +/- buttons too.

I also think it could be useful to allow an option for a date field to not show the time at all if that's not relevant to the use-case.  I've built things over the years where the time was just a distraction and I had to add something along the lines of "Ignore the time" in the field instructions.  I suppose in these cases the timestamp in the Database would just be midnight of the chosen day.

**Comment by PaulBaileyAtChan** [comment]
+1 No strong opinion on the exact UI, but being able to set the date using the calendar pop-up, but not the time, doesn't make a lot of sense, so adding a nifty tool for that would be a lot more friendly.

Like @pyrobob83, I also find myself often only wanting/using part of the date/time combo (sometimes just day/month, and not even the year), and having the other pieces is a distraction. Sounds like a tricky bit of UI to get right, but anything along those lines would be useful.

Given the above, I'm wondering if a useful extension to the template language for dates would be to allow individual components to be picked out, something like (for e.g.):

```
{date_field:year}
{date_field:month}
{date_field:day}

```

The above are all possible using the existing formatting codes, natch, but those are really for formatted output, and it can be clunky to use them for conditional logic in templates:

`{if date_field:year == 2023}`

would be simpler/clearer than:

`{if '{date_field format="%Y"}' == '2023'}`

**Comment by chrisdaviesweb** [comment]
+1! I have multiple clients that output timeslots - it's always been a bit of a strain to explain that the date field doesn't allow time selection and that it has to be done manually. This would be a big help in these kind of scenarios.

**Comment by 620studio** [comment]
I'm just now coming across this in a project and having a time picker option would help. Otherwise, I'm having to load a separate library (jQuery UI) just for better UX
